### ISSUE TITLE
Check for "None" return type

### ISSUE THREAD

**Issue by wizzardx** [agronholm/typeguard#5]
This code passes runtime checks, but should fail:

```python
#!/usr/bin/env python3

from typeguard import typechecked


@typechecked
def main() -> None:
    return 'asdf'


if __name__ == '__main__':
    main()
```

Main is defined as having no return type according to PEP 484:

https://www.python.org/dev/peps/pep-0484/#using-none

Runtime type checking does work if you define main like this:

```python
@typechecked
def main() -> type(None):
    return 'asdf'
```

**Comment by agronholm** [comment]
Fixed in 7406e8985c506d7d62e3beda39be9d8dc652b351.

**Comment by agronholm** [comment]
FYI I've cut a release that includes the fix.
